### DISCUSSION TITLE
Ginga on MacOSX Big Sur

### DISCUSSION THREAD

**Discussion by zakamska** [ejeschke/ginga#989] [Q&A] [answered]
Dear Ginga Developers, 

I am new to Ginga, trying to use it as part of PypeIt (Prochaska et al.). I am running MacOSX 11.4 (Big Sur), Python 3.10.2 and Ginga 3.2.0. 

I cannot seem to get the graphics working. Any attempts to use QT result in an error: 
`qt.qpa.drawing: Layer-backing can not be explicitly controlled on 10.14 when built against the 10.14 SDK`
and / or
```
Traceback (most recent call last):
  File "/Applications/anaconda3/envs/pypeit/lib/python3.10/site-packages/ginga/qtw/Widgets.py", line 227, in sizeChange
    self.setMaximumHeight(docHeight)
TypeError: setMaximumHeight(self, int): argument 1 has unexpected type 'float'
```

I tried a clean install of ginga and various installs of QT, still nothing (either one of the errors above or `UI toolkit choose error: qt or gtk variants not found`

I tried a clean install of ginga with GTK3, I get `Could not build wheels for pycairo which use PEP 517 and cannot be installed directly`

I've googled these issues, and it seems they are known (QT doesn't play well with Mac 10.14 and above), but I have not seen any solutions. Does anybody have any suggestions? Many thanks.

**Comment by ejeschke** [comment]
Hi @zakamska, I ran Ginga just fine on Mac OS X "Big Sur" (currently upgraded to "Monterey" and still running fine).  Are you using an Anaconda or Miniconda environment (the preferred way for most folks)?

**Comment by ejeschke** [comment]
Sorry, I see from the traceback now that you are using anaconda.

**Comment by ejeschke** [comment, marked as answer]
When I try to build a conda environment for ginga using python 3.10 I get dependence conflicts:
```bash
$ conda create -n py310 python=3.10
...
...
$ conda activate py310
(py310) macbook-pro-5: eric$ conda install scipy matplotlib pyqt qtpy pillow astropy 
...
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: \ 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                            |  

UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - pyqt -> python[version='>=2.7,<2.8.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.6,<3.7.0a0|>=3.5,<3.6.0a0|>=3.7,<3.8.0a0']
  - qtpy -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0']

Your python: python=3.10

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.

The following specifications were found to be incompatible with each other:
...
(followed by a lot more package conflicts)
```
I am not sure that all of the necessary packages for running Qt successfully are available using the standard conda repos. 

Could you try creating a separate python 3.9 environment and installing the necessary packages as shown above? Then "pip install ginga".

**Reply by zakamska** [reply]
YES!! This worked. Thank you. Since I am unfamiliar with ginga, I can't confirm that everything is working, but I am seeing the graphic window for the first time. THANK YOU. 

I'll make sure I downgrade to 3.9 in the other applications.

**Comment by ejeschke** [comment]
Excellent. Have a great day!
